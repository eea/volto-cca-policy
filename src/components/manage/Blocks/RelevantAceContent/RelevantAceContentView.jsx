import React from 'react';
import { List } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { flattenToAppURL } from '@plone/volto/helpers';

const RelevantAceContentView = (props) => {
  const { data } = props;
  const { title, _v_results, items } = data;
  const results = _v_results || [];

  // console.log(data);

  return (
    <div className="block relevant-acecontent-block">
      {title && <h4>{title}</h4>}
      {items && items.length > 0 ? (
        <>
          {items.map((item, index) => (
            <List.Item key={index}>
              <Link to={flattenToAppURL(item.link)}>{item.item_title}</Link>
            </List.Item>
          ))}
        </>
      ) : (
        <>
          {results.map((result, index) => (
            <List.Item key={index} title={result[1]}>
              <Link to={flattenToAppURL(result[4])}>{result[0]}</Link>
            </List.Item>
          ))}
        </>
      )}
    </div>
  );
};

export default RelevantAceContentView;
