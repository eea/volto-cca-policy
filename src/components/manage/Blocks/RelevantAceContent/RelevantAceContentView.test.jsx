import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import RelevantAceContentView from './RelevantAceContentView';

const mockStore = configureStore();

jest.mock('semantic-ui-react', () => ({
  ...jest.requireActual('semantic-ui-react'),
  Icon: () => <div>Icon</div>,
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => <div>Link</div>,
}));

describe('RelevantAceContentView', () => {
  it('should render the component', () => {
    const data = {
      _v_results: [
        ['Publications and reports', 'Publications and reports', '/', []],
        ['Information portals', 'Information portals', '/', []],
      ],
    };

    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <RelevantAceContentView data={data} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
