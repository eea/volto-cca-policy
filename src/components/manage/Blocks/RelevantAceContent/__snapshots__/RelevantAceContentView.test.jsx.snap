// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelevantAceContentView should render the component 1`] = `
<div
  className="block relevant-acecontent-block"
>
  <div
    className="ui list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
      title="Publications and reports"
    >
      <i
        aria-hidden="true"
        className="angle right icon"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        <div>
          Link
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
      title="Information portals"
    >
      <i
        aria-hidden="true"
        className="angle right icon"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        <div>
          Link
        </div>
      </div>
    </div>
  </div>
</div>
`;
